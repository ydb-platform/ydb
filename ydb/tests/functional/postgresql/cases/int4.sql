--
-- INT4
--
CREATE TABLE INT4_TBL(f1 int4 primary key);
INSERT INTO INT4_TBL(f1) VALUES ('   0  ');
INSERT INTO INT4_TBL(f1) VALUES ('123456     ');
INSERT INTO INT4_TBL(f1) VALUES ('    -123456');
INSERT INTO INT4_TBL(f1) VALUES ('34.5');
-- largest and smallest values
INSERT INTO INT4_TBL(f1) VALUES ('2147483647');
INSERT INTO INT4_TBL(f1) VALUES ('-2147483647');
-- bad input values -- should give errors
INSERT INTO INT4_TBL(f1) VALUES ('1000000000000');
INSERT INTO INT4_TBL(f1) VALUES ('asdf');
INSERT INTO INT4_TBL(f1) VALUES ('     ');
INSERT INTO INT4_TBL(f1) VALUES ('   asdf   ');
INSERT INTO INT4_TBL(f1) VALUES ('- 1234');
INSERT INTO INT4_TBL(f1) VALUES ('123       5');
INSERT INTO INT4_TBL(f1) VALUES ('');
SELECT * FROM INT4_TBL;
SELECT i.* FROM INT4_TBL i WHERE i.f1 <> int2 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 <> int4 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 = int2 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 = int4 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 < int2 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 < int4 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 <= int2 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 <= int4 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 > int2 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 > int4 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 >= int2 '0';
SELECT i.* FROM INT4_TBL i WHERE i.f1 >= int4 '0';
-- positive odds
SELECT i.* FROM INT4_TBL i WHERE (i.f1 % int2 '2') = int2 '1';
-- any evens
SELECT i.* FROM INT4_TBL i WHERE (i.f1 % int4 '2') = int2 '0';
SELECT i.f1, i.f1 * int2 '2' AS x FROM INT4_TBL i;
SELECT i.f1, i.f1 * int2 '2' AS x FROM INT4_TBL i
WHERE abs(f1) < 1073741824;
SELECT i.f1, i.f1 * int4 '2' AS x FROM INT4_TBL i;
SELECT i.f1, i.f1 * int4 '2' AS x FROM INT4_TBL i
WHERE abs(f1) < 1073741824;
SELECT i.f1, i.f1 + int2 '2' AS x FROM INT4_TBL i;
SELECT i.f1, i.f1 + int2 '2' AS x FROM INT4_TBL i
WHERE f1 < 2147483646;
SELECT i.f1, i.f1 + int4 '2' AS x FROM INT4_TBL i;
SELECT i.f1, i.f1 + int4 '2' AS x FROM INT4_TBL i
WHERE f1 < 2147483646;
SELECT i.f1, i.f1 - int2 '2' AS x FROM INT4_TBL i;
SELECT i.f1, i.f1 - int2 '2' AS x FROM INT4_TBL i
WHERE f1 > -2147483647;
SELECT i.f1, i.f1 - int4 '2' AS x FROM INT4_TBL i;
SELECT i.f1, i.f1 - int4 '2' AS x FROM INT4_TBL i
WHERE f1 > -2147483647;
SELECT i.f1, i.f1 / int2 '2' AS x FROM INT4_TBL i;
SELECT i.f1, i.f1 / int4 '2' AS x FROM INT4_TBL i;
