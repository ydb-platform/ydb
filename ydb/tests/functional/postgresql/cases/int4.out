--
-- INT4
--
CREATE TABLE INT4_TBL(f1 int4 primary key);
INSERT INTO INT4_TBL(f1) VALUES ('   0  ');
INSERT INTO INT4_TBL(f1) VALUES ('123456     ');
INSERT INTO INT4_TBL(f1) VALUES ('    -123456');
INSERT INTO INT4_TBL(f1) VALUES ('34.5');
ERROR:  invalid input syntax for type integer: "34.5"
LINE 1: INSERT INTO INT4_TBL(f1) VALUES ('34.5');
                                         ^
-- largest and smallest values
INSERT INTO INT4_TBL(f1) VALUES ('2147483647');
INSERT INTO INT4_TBL(f1) VALUES ('-2147483647');
-- bad input values -- should give errors
INSERT INTO INT4_TBL(f1) VALUES ('1000000000000');
ERROR:  value "1000000000000" is out of range for type integer
LINE 1: INSERT INTO INT4_TBL(f1) VALUES ('1000000000000');
                                         ^
INSERT INTO INT4_TBL(f1) VALUES ('asdf');
ERROR:  invalid input syntax for type integer: "asdf"
LINE 1: INSERT INTO INT4_TBL(f1) VALUES ('asdf');
                                         ^
INSERT INTO INT4_TBL(f1) VALUES ('     ');
ERROR:  invalid input syntax for type integer: "     "
LINE 1: INSERT INTO INT4_TBL(f1) VALUES ('     ');
                                         ^
INSERT INTO INT4_TBL(f1) VALUES ('   asdf   ');
ERROR:  invalid input syntax for type integer: "   asdf   "
LINE 1: INSERT INTO INT4_TBL(f1) VALUES ('   asdf   ');
                                         ^
INSERT INTO INT4_TBL(f1) VALUES ('- 1234');
ERROR:  invalid input syntax for type integer: "- 1234"
LINE 1: INSERT INTO INT4_TBL(f1) VALUES ('- 1234');
                                         ^
INSERT INTO INT4_TBL(f1) VALUES ('123       5');
ERROR:  invalid input syntax for type integer: "123       5"
LINE 1: INSERT INTO INT4_TBL(f1) VALUES ('123       5');
                                         ^
INSERT INTO INT4_TBL(f1) VALUES ('');
ERROR:  invalid input syntax for type integer: ""
LINE 1: INSERT INTO INT4_TBL(f1) VALUES ('');
                                         ^
SELECT * FROM INT4_TBL;
     f1      
-------------
 -2147483647
     -123456
           0
      123456
  2147483647
(5 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 <> int2 '0';
     f1      
-------------
 -2147483647
     -123456
      123456
  2147483647
(4 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 <> int4 '0';
     f1      
-------------
 -2147483647
     -123456
      123456
  2147483647
(4 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 = int2 '0';
 f1 
----
  0
(1 row)

SELECT i.* FROM INT4_TBL i WHERE i.f1 = int4 '0';
 f1 
----
  0
(1 row)

SELECT i.* FROM INT4_TBL i WHERE i.f1 < int2 '0';
     f1      
-------------
 -2147483647
     -123456
(2 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 < int4 '0';
     f1      
-------------
 -2147483647
     -123456
(2 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 <= int2 '0';
     f1      
-------------
 -2147483647
     -123456
           0
(3 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 <= int4 '0';
     f1      
-------------
 -2147483647
     -123456
           0
(3 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 > int2 '0';
     f1     
------------
     123456
 2147483647
(2 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 > int4 '0';
     f1     
------------
     123456
 2147483647
(2 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 >= int2 '0';
     f1     
------------
          0
     123456
 2147483647
(3 rows)

SELECT i.* FROM INT4_TBL i WHERE i.f1 >= int4 '0';
     f1     
------------
          0
     123456
 2147483647
(3 rows)

-- positive odds
SELECT i.* FROM INT4_TBL i WHERE (i.f1 % int2 '2') = int2 '1';
     f1     
------------
 2147483647
(1 row)

-- any evens
SELECT i.* FROM INT4_TBL i WHERE (i.f1 % int4 '2') = int2 '0';
   f1    
---------
 -123456
       0
  123456
(3 rows)

SELECT i.f1, i.f1 * int2 '2' AS x FROM INT4_TBL i;
ERROR:  integer out of range
SELECT i.f1, i.f1 * int2 '2' AS x FROM INT4_TBL i
WHERE abs(f1) < 1073741824;
   f1    |    x    
---------+---------
 -123456 | -246912
       0 |       0
  123456 |  246912
(3 rows)

SELECT i.f1, i.f1 * int4 '2' AS x FROM INT4_TBL i;
ERROR:  integer out of range
SELECT i.f1, i.f1 * int4 '2' AS x FROM INT4_TBL i
WHERE abs(f1) < 1073741824;
   f1    |    x    
---------+---------
 -123456 | -246912
       0 |       0
  123456 |  246912
(3 rows)

SELECT i.f1, i.f1 + int2 '2' AS x FROM INT4_TBL i;
ERROR:  integer out of range
SELECT i.f1, i.f1 + int2 '2' AS x FROM INT4_TBL i
WHERE f1 < 2147483646;
     f1      |      x      
-------------+-------------
 -2147483647 | -2147483645
     -123456 |     -123454
           0 |           2
      123456 |      123458
(4 rows)

SELECT i.f1, i.f1 + int4 '2' AS x FROM INT4_TBL i;
ERROR:  integer out of range
SELECT i.f1, i.f1 + int4 '2' AS x FROM INT4_TBL i
WHERE f1 < 2147483646;
     f1      |      x      
-------------+-------------
 -2147483647 | -2147483645
     -123456 |     -123454
           0 |           2
      123456 |      123458
(4 rows)

SELECT i.f1, i.f1 - int2 '2' AS x FROM INT4_TBL i;
ERROR:  integer out of range
SELECT i.f1, i.f1 - int2 '2' AS x FROM INT4_TBL i
WHERE f1 > -2147483647;
     f1     |     x      
------------+------------
    -123456 |    -123458
          0 |         -2
     123456 |     123454
 2147483647 | 2147483645
(4 rows)

SELECT i.f1, i.f1 - int4 '2' AS x FROM INT4_TBL i;
ERROR:  integer out of range
SELECT i.f1, i.f1 - int4 '2' AS x FROM INT4_TBL i
WHERE f1 > -2147483647;
     f1     |     x      
------------+------------
    -123456 |    -123458
          0 |         -2
     123456 |     123454
 2147483647 | 2147483645
(4 rows)

SELECT i.f1, i.f1 / int2 '2' AS x FROM INT4_TBL i;
     f1      |      x      
-------------+-------------
 -2147483647 | -1073741823
     -123456 |      -61728
           0 |           0
      123456 |       61728
  2147483647 |  1073741823
(5 rows)

SELECT i.f1, i.f1 / int4 '2' AS x FROM INT4_TBL i;
     f1      |      x      
-------------+-------------
 -2147483647 | -1073741823
     -123456 |      -61728
           0 |           0
      123456 |       61728
  2147483647 |  1073741823
(5 rows)
